node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric.go

```diff
@@ -155,20 +155,22 @@ func (r OsDeviceConnectivityHelperScsiGeneric) GetMpathDevice(volumeId string, l
 	}
 	var devicePaths []string
 	var errStrings []string
-	var targetPath string
+	var subsystemPrefix string
 
 	if connectivityType == ConnectionTypeFC {
-		targetPath = fmt.Sprintf("/dev/disk/by-path/%s*", fcSubsystem)
+		subsystemPrefix = fcSubsystemPrefix
 		// In host, the path like this: /dev/disk/by-path/pci-0000:13:00.0-fc-0x500507680b25c0aa-lun-0
 		// So add prefix "0x" for the arrayIdentifiers
 		for index, wwn := range arrayIdentifiers {
 			arrayIdentifiers[index] = "0x" + strings.ToLower(wwn)
 		}
 	}
 	if connectivityType == ConnectionTypeISCSI {
-		targetPath = "/dev/disk/by-path/ip*"
+		subsystemPrefix = "ip*-"
 	}
 
```

Comment:
leave devicePath instance variable as before and init it here. Don't define it here as local variable.
What is the reason u devicePath as instance variable?

Reply:
I'm not sure I understand what you mean.
could you clarify?

Reply:
I mean targetPath
Leave it as instance variable
and don't add subsystemPrefix  as instance variable

Reply:
why?
and are you implying `targetPath` is not an instance variable in my current change? if so, why?

Reply:
My mistake sorry. They are not instance variables.